core/src/test/java/org/apache/iceberg/avro/AvroTestHelpers.java

```diff
@@ -82,7 +82,7 @@ static void assertEquals(Types.StructType struct, Record expected, Record actual
   static void assertEquals(Types.ListType list, List<?> expected, List<?> actual) {
     Type elementType = list.elementType();
 
-    Assert.assertEquals("List size should match", expected.size(), actual.size());
+    org.junit.jupiter.api.Assertions.assertEquals(expected.size(), actual.size(),"List size should match");
```

Comment:
Based on the conversations in https://github.com/apache/iceberg/issues/7160 I think we'd rather want to use AssertJ assertions